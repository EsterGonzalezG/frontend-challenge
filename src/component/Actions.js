import React from 'react';

export const Actions = ({ product, setColor, setStorage }) => (
  <div>
    {
      <form action=''>
        <div className='color_container'>
          <label htmlFor='color'>Color:</label>
          <select name='color' id='color' value={product.colors} onChange={(event) => setColor(event.target.value)}>
            {product.options
              ? product.options.colors.map((color, index) =>
                  index === 0 ? (
                    <option value={color.code} key={index} defaultValue={color.name}>
                      {color.name}
                    </option>
                  ) : (
                    <option value={color.code} key={index}>
                      {color.name}
                    </option>
                  ),
                )
              : ''}
          </select>
        </div>
        <div className='storage_container'>
          <label htmlFor='storage'>Almacenamiento:</label>
          <select
            name='storage'
            id='storage'
            value={product.storage}
            onChange={(event) => setStorage(event.target.value)}
          >
            {product.options
              ? product.options.storages.map((storage, index) =>
                  index === 0 ? (
                    <option value={storage.code} key={index} defaultValue={storage.name}>
                      {storage.name}
                    </option>
                  ) : (
                    <option value={storage.code} key={index}>
                      {storage.name}
                    </option>
                  ),
                )
              : ''}
          </select>
        </div>
      </form>
    }
    <button>AÃ±adir</button>
  </div>
);
