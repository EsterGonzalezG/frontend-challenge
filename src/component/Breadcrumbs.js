import React from 'react';
import { Link, useLocation } from 'react-router-dom';

export const Breadcrumbs = React.memo(() => {
  const location = useLocation();
  const config = {
    home: [{ label: 'Home', path: '/' }],
    details: [{ label: 'Home', path: '/' }, { label: 'Product details' }],
  };

  function crumb() {
    if (location.pathname === '/') {
      return renderCrumb(config.home);
    } else {
      return renderCrumb(config.details);
    }
  }

  function renderCrumb(option) {
    return option.map(({ label, path }, index) => {
      return option.length - 1 === index ? (
        <li className='breadcrumbs-item' key={index}>
          {label}
        </li>
      ) : (
        <li key={index}>
          <Link to={path} className='breadcrumbs-link'>
            {label}
          </Link>
        </li>
      );
    });
  }

  return (
    <nav>
      <ul className='breadcrumbs color-dark-60'>{crumb()}</ul>
    </nav>
  );
});
