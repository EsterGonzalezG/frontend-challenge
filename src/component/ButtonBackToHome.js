import { PropTypes } from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

export const ButtonBackToHome = ({ children }) => (
  <Link className='' to='/'>
    {children}
  </Link>
);
ButtonBackToHome.prototype = {
  children: PropTypes.string.isRequired,
};
