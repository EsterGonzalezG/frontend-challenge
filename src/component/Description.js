import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { setTitleDescription, Subtitle } from './index';

export const Description = React.memo(({ description }) => {
  const [descriptionValue, setDescriptionValue] = useState([]);
  useEffect(() => {
    setDescriptionValue(setTitleDescription(description));
  }, [description]);

  return (
    <>
      <Subtitle>Characteristics:</Subtitle>
      <ul className='color-dark-100 l-marginBottom-24 font-m'>
        {descriptionValue.map(({ id, value }) => (
          <li key={id} className='color-violet'>
            {id} :
            {value !== '' ? <span className='color-dark-60'>{value}</span> : <span className='color-dark-60'> - </span>}
          </li>
        ))}
      </ul>
    </>
  );
});

Description.prototype = {
  description: PropTypes.shape({
    brand: PropTypes.string,
    model: PropTypes.string,
    price: PropTypes.string,
    cpu: PropTypes.string,
    ram: PropTypes.string,
    os: PropTypes.string,
    displayResolution: PropTypes.string,
    battery: PropTypes.string,
    primaryCamera: PropTypes.string,
    secondaryCmera: PropTypes.string,
    dimentions: PropTypes.string,
    weight: PropTypes.string,
  }).isRequired,
};
