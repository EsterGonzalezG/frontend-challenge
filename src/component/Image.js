import React from "react";
import PropTypes from 'prop-types';


export const Image = ({ item }) => (
    <img loading="lazy" className='l-marginBottom-16' src={item.imgUrl} alt={item.title}/>
)

Image.prototype = {
    imgUrl: PropTypes.string,
    title: PropTypes.string
}