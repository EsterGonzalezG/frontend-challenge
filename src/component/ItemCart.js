import { PropTypes } from 'prop-types';
import React from 'react';
import { BiCartAlt } from 'react-icons/bi';

export const ItemCard = ({ cart }) => (
  <div className='itemCart l-flexCenterAround'>
    <span className='itemCart-icon'>
      <BiCartAlt />
    </span>
    <p className='itemCart-number' data-number={cart}></p>
  </div>
);
ItemCard.prototype = {
  cart: PropTypes.number,
};
