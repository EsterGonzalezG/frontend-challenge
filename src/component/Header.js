import React from 'react';

import { Title } from './Title';
import { Breadcrumbs } from './Breadcrumbs';
import {ItemCard}  from './ItemCart'
export const Header = () => (
    <div className='header l-marginBottom-24'>
        <div className='header-content'>
        <Title>Tienda de mÃ³biles</Title>
        <Breadcrumbs/>
        </div>
        <ItemCard/>
    </div>
)


