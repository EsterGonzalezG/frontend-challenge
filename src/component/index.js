import { setTitleDescription } from '../utils/setTitleDescription';
import { CartContext } from './../useContext/CartContext';
import { Breadcrumbs } from './Breadcrumbs';
import { ButtonBackToHome } from './ButtonBackToHome';
import { Image } from './Image';
import { ItemCard } from './ItemCart';
import { NotData } from './NotData';
import { Search } from './Search';
import { Select } from './Select';
import { Title } from './Title';

export {
  ButtonBackToHome,
  Select,
  Breadcrumbs,
  ItemCard,
  Image,
  Search,
  CartContext,
  NotData,
  Title,
  setTitleDescription,
};
