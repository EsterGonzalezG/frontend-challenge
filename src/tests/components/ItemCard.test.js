import { mount } from 'enzyme';
import React from 'react';
import { ItemCard } from '../../component/ItemCart';
import { CartContext } from '../../useContext/CartContext';
describe('Test ItemCard component', () => {
  const cart = 1;
  const wrapper = mount(
    <CartContext.Provider value={cart}>
      <ItemCard />
    </CartContext.Provider>,
  );
  test('should be render a component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
