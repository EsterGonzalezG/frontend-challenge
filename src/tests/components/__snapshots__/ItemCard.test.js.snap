// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemCard component should be render a component 1`] = `
<div
  className="itemCart l-flexCenterAround"
>
  <span
    className="itemCart-icon"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.246 11L18.246 11 19.675 6 6.428 6z"
        fill="none"
        key="0"
      />
      <path
        d="M21,4H5H4H2v2h2h0.3l3.282,9.025C8.011,16.206,9.145,17,10.401,17H19v-2h-8.599c-0.419,0-0.797-0.265-0.94-0.658L8.973,13 h9.273c0.89,0,1.68-0.598,1.923-1.451l1.793-6.274c0.086-0.302,0.025-0.626-0.163-0.877C21.609,4.147,21.313,4,21,4z M18.246,11 h-10L6.428,6h13.247L18.246,11z"
        key="1"
      />
      <circle
        cx="10.5"
        cy="19.5"
        key="2"
        r="1.5"
      />
      <circle
        cx="16.5"
        cy="19.5"
        key="3"
        r="1.5"
      />
    </svg>
  </span>
  <p
    className="itemCart-number"
  />
</div>
`;
